x-pack/libbeat/management/manager.go

```diff
@@ -112,29 +112,36 @@ func (cm *ConfigManager) worker() {
 	defer cm.wg.Done()
 
 	// Initial fetch && apply (even if errors happen while fetching)
-	cm.fetch()
-	cm.apply()
+	firstRun := true
+	period := 0 * time.Second
```

Comment:
Not directly related to this change, but we could add an initial random delay, like with the [`splay` option in puppet.](https://stackoverflow.com/a/32908565/28855)

Reply:
This is a good idea, thank you for bringing it up, it was discussed initially but we never go back to it, will try to open a new PR for it

Reply:
Oh ok, so the difference is that now it can also trigger a Save on first run.